src/os_driver_nodelet.cpp

```diff
@@ -38,18 +38,66 @@ class OusterDriver : public OusterSensor {
         halt();
     }
 
+   protected:
+    virtual void onInit() override {
+        create_imu_pub();
+        create_point_cloud_pubs();
+        create_laser_scan_pubs();
+        create_image_pubs();
+        OusterSensor::onInit();
+    }
+
    private:
 
     virtual void on_metadata_updated(const sensor::sensor_info& info) override {
         OusterSensor::on_metadata_updated(info);
-
         // for OusterDriver we are going to always assume static broadcast
         // at least for now
         tf_bcast.parse_parameters(getPrivateNodeHandle());
         tf_bcast.broadcast_transforms(info);
+        create_handlers();
+    }
+
+    void create_imu_pub() {
+        imu_pub = getNodeHandle().advertise<sensor_msgs::Imu>("imu", 100);
+    }
+
+    void create_point_cloud_pubs() {
+        // NOTE: always create the 2nd topic
+        lidar_pubs.resize(2);
+        for (int i = 0; i < 2; ++i) {
+            lidar_pubs[i] = getNodeHandle().advertise<sensor_msgs::PointCloud2>(
+                topic_for_return("points", i), 10);
+        }
+    }
+
+    void create_laser_scan_pubs() {
+        // NOTE: always create the 2nd topic
```

Comment:
We used to create the 2nd topics only if the current profile supported DUAL returns, however, since the user can potentially switch between different profile on reconnection it wouldn't bound well that we create and destroy topics when switching between dual vs non-dual for downstream consumers.